File: 01_Leetcode_FAANG/Animesh Nayak/Single Number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int c=0;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size();i++)
            c=c+nums[i]*(pow(-1,i%2));
        return c;
    }
};

File: 01_Leetcode_FAANG/Single_number.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for n in nums:
            ans ^= n
        return ans
        
ans = Solution()
print(ans.singleNumber([4,1,2,1,2,3,3]))
